### PR TITLE
Change `RequestInterface::getCurlPostFields()` return type from `?array` to `array`

### PR DESCRIPTION
**Pull request by OskarStark**
Technically a BC break, but only if you created your own `Request` class, so low chance

### CODE REVIEW

File: tests/Client/Request/RequestTest.php [hunk 1]
```diff
@@ -42,10 +42,4 @@ public function testGetApiUrl(Request $request): void
     {
         $this->assertEquals('https://test.com/api', $request->getApiUrl());
     }
-
-    #[Depends('testCanBeCrated')]
-    public function testGetCurlPostFields(Request $request): void
-    {
-        $this->assertIsArray($request->getCurlPostFields());
-    }
```
Comment (OskarStark) [inline tests/Client/Request/RequestTest.php:50]: not needed anymore, at is always an `array`

File: src/Client/Request/Request.php [hunk 2]
```diff
@@ -33,14 +33,14 @@ class Request implements RequestInterface
...
-     * @param null|array<string, string> $curlPostFields array for CURLOPT_POSTFIELDS curl argument
+     * @param string                $apiUrl         Full API URL
+     * @param string                $method         Either GET or POST
+     * @param array<string, string> $curlPostFields array for CURLOPT_POSTFIELDS curl argument
      */
     public function __construct(
         private readonly string $apiUrl,
         private readonly string $method,
-        private readonly ?array $curlPostFields = null,
+        private readonly ?array $curlPostFields = [],
```
Comment (slunak) [inline src/Client/Request/Request.php:43]: Here no need for `?`, just `array`?

Comment (OskarStark) [reply src/Client/Request/Request.php]: > Here no need for ?, just array?

Good catch

> Maybe we can review all instances of $request = new Request($client->buildApiUrl(), Request::GET); and add [] as a third argument?

We can 👍

**Review by slunak — APPROVED** [review]
Thank you!

**Pull request merged**